grails-app/services/org/pih/warehouse/dashboard/IndicatorDataService.groovy

```diff
@@ -1168,6 +1169,83 @@ class IndicatorDataService {
         return getBackdatedShipmentsChart(locationId, monthsLimit, StockMovementDirection.INBOUND)
     }
 
+    @Cacheable(value = "dashboardCache", key = { "getItemsWithBackdatedShipments-${location?.id}" })
+    GraphData getItemsWithBackdatedShipments(Location location) {
+        Integer monthsLimit = Holders.config.openboxes.dashboard.backdatedShipments.monthsLimit
+        Date timeLimit = LocalDate.now().minusMonths(monthsLimit).toDate()
+        Map<String, Integer> amountOfItemsWithBackdatedShipments = [
+            (Constants.ONE)            : 0,
+            (Constants.TWO)            : 0,
+            (Constants.THREE)          : 0,
+            (Constants.FOUR_OR_MORE)   : 0,
+        ]
+
+        String query = '''
+            SELECT ii.product_id, COUNT(ii.product_id) as products, GROUP_CONCAT(bt.shipment_number SEPARATOR ' ') as shipments, DATE_FORMAT(sc.last_stock_count, "%d-%M-%Y") FROM (
+                SELECT t.id as transaction_id, s.shipment_number FROM shipment s
+                INNER JOIN `transaction` t ON t.outgoing_shipment_id  = s.id
+                WHERE DATE_ADD(t.transaction_date, INTERVAL :daysOffset DAY) < t.date_created
+                AND t.date_created > :timeLimit
+                AND s.origin_id = :locationId
+            ) AS bt
+            INNER JOIN transaction_entry te ON te.transaction_id = bt.transaction_id
+            INNER JOIN inventory_item ii ON te.inventory_item_id = ii.id
+            LEFT JOIN (
+                SELECT ii.product_id, MAX(t.transaction_date) as last_stock_count from transaction_entry te 
+                LEFT JOIN inventory_item ii ON te.inventory_item_id = ii.id 
+                LEFT JOIN `transaction` t ON t.id = te.transaction_id
+                LEFT JOIN transaction_type tt ON tt.id = t.transaction_type_id 
+                WHERE t.inventory_id = :inventoryId
+                AND tt.transaction_code = :transactionCode
+                GROUP BY ii.product_id 
+            ) as sc ON sc.product_id = ii.product_id
+            GROUP BY ii.product_id, sc.last_stock_count
+        '''
+
+        List<GroovyRowResult> results = dataService.executeQuery(query, [
+                locationId: location?.id,
+                inventoryId: location?.inventory?.id,
+                transactionCode: TransactionCode.PRODUCT_INVENTORY.name(),
+                daysOffset: Holders.config.openboxes.dashboard.backdatedShipments.daysOffset,
+                timeLimit: timeLimit,
+        ])
+
+        List<TableData> tableData = results.collect { new TableData(
+                it[0] as String,          // Item
+                it[2] as String,          // Shipments
+                it[3] as String ?: "None" // Last count
```

Comment:
I think "Never" would fit here more, unless "None" was requested, then you can disregard this comment

Reply:
it was requested

Reply:
@awalkowiak I guess this is the sort of detail we need to catch during "design".

One additional point (add to #3849)

> [!IMPORTANT] 
> **Convention** Don't use "magic strings" 
> Source: https://deviq.com/antipatterns/magic-strings

1. We should use an enum if there are multiple coded classification values (i.e. Never, Long Ago, Somewhat Recently, Recently) 
1. We should use a Constant if we need to support just a string literal in logic (i.e. Default, Never, None). 
1. We should allow constants and enums to be i18n'd, so there's a way to translate the actual string literal. 

Add it right under the default constant and call it never. 
```
static final String DEFAULT = "DEFAULT"
static final String NEVER = "NEVER"
```

Reply:
@jmiranda 
> I guess this is the sort of detail we need to catch during "design".

This particular label did not appear in the ticket at all, and it was just suggested by Manon outside the ticket. So it is hard to speak about design here if it was missed since the beginning. I am also not sure if I would spot it even with a proper dev design because I would be focusing on the SQL query itself, it is hard to tell.

Reply:
(added a topic around label and messages designing for a tech huddle)

Reply:
FWIW, in case it wasn't clear ... this was not criticism of what we should have done here but rather what patterns we should be looking for when we start to do proper design in the near future.